module/database/FileDatabase.py

```diff
@@ -831,7 +832,9 @@ def getFile(self, id):
         self.c.execute("SELECT url, name, size, status, error, plugin, package, linkorder FROM links WHERE id=?", (str(id), ))
         r = self.c.fetchone()
         if not r: return None
-        return PyFile(self.manager, id, * r)
+        pyfile = PyFile(self.manager, id, * r)
```

Comment:
Admittedly have to mention that I don't know whether this is the right place to re-init the `pyfile.plugin`.
Maybe there is a better place to put in order to remain DRY. But it's a fact that in some place there has to be made sure that `pyfile.plugin` should never be `None` but set to the appropriate plugin.